print_job_queue/models.py

```diff
@@ -0,0 +1,80 @@
+from enum import Enum, unique
+
+from django.conf import settings
+from django.db import models
+
+
+@unique
+class PrintJobState(Enum):
+    # The job has been created and is ready for pickup.
+    PENDING = 0
+
+    # The job is being processed by a worker.
+    PROCESSING = 1
+
+    # The job has been processed by a worker. This is a terminal state.
+    DONE = 2
+
+
+class PrintJob(models.Model):
+    STATE = (
+        (member.value, name)
+        for name, member in PrintJobState.__members__.items()
+    )
+    
+    job_id = models.AutoField(primary_key=True)
+
+    # See PrintJobState.
+    state = models.IntegerField(choices=STATE)
+
+    # Whether the job is still valid. An invalid job should be considered
+    # defunct. Prefer flipping this to false to deleting an instance from the
+    # database.
+    is_valid = models.BooleanField(default=True)
+
+    # The worker that is handling this job. Should be set when the job is in
+    # either PROCESSING or DONE state.
+    handling_worker = models.CharField(max_length=25, blank=True)
```

Comment:
Super duper nit: I think we can simply call this `worker`.

Reply:
Done.